gobblin-core/src/main/java/gobblin/source/extractor/extract/sftp/SftpFsHelper.java

```diff
@@ -51,10 +55,52 @@ public SftpFsHelper(State state) {
     this.state = state;
   }
 
-  public ChannelSftp getSftpConnection() {
-    return this.channelSftp;
+  /**
+   * Create new channel every time a command needs to be executed. This is required to support execution of multiple
+   * commands in parallel. All created channels are cleaned up when the session is closed.
+   * @see #cleanupSftpChannels()
+   * @see #close()
+   *
+   * @return a new {@link ChannelSftp}
+   * @throws SftpException
+   */
+  public ChannelSftp getSftpChannel() throws SftpException {
```

Comment:
Right now I am explicitly calling channel.disconnect after I finish using a channel. Yes TryWithResource is the ideal way. I would want to do the java upgrade in a different PR.